lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -77,17 +78,30 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
     if (request.query && request.query['x-amz-acl']) {
         headers['x-amz-acl'] = request.query['x-amz-acl'];
     }
-    const metadataStoreParams = {
-        objectKey, authInfo, metaHeaders, size,
-        contentType: request.headers['content-type'],
-        cacheControl: request.headers['cache-control'],
-        contentDisposition: request.headers['content-disposition'],
-        contentEncoding: removeAWSChunked(request.headers['content-encoding']),
-        expires: request.headers.expires,
-        headers,
-        log,
-        isDeleteMarker,
-    };
+    if (isDeleteMarker) {
```

Comment:
can't you define metadataStoreParams with all of these items and if !isDeleteMarker add the other items (rather than starting from scratch each time)?